src/main/java/com/ait/lienzo/client/core/shape/wires/WiresManager.java

```diff
@@ -157,11 +162,20 @@ public WiresShapeControl register(final WiresShape shape, final boolean addIntoI
 
         if (addIntoIndex)
         {
-
             // Shapes added to the align and distribute index.
             final AlignAndDistributeControl alignAndDistrControl = addToIndex(shape);
             handler.setAlignAndDistributeControl(alignAndDistrControl);
 
+            shape.addWiresResizeEndHandler(new WiresResizeEndHandler()
+            {
+                @Override
+                public void onShapeResizeEnd(WiresResizeEndEvent event)
+                {
+                    removeFromIndex(shape);
+                    final AlignAndDistributeControl controls = addToIndex(shape);
+                    handler.setAlignAndDistributeControl(controls);
+                }
+            });
         }
```

Comment:
This is the fix, the rest changes just a refactoring for easier understanding what's going on around here.